File: packages/motion_sleep/lib/src/entities/sleep_time.dart
part of motion_sleep;

class SleepTime {
  final int startingHours;
  final int startingMinutes;
  final int endingHours;
  final int endingMinutes;

  SleepTime(
    this.startingHours,
    this.startingMinutes,
    this.endingHours,
    this.endingMinutes,
  );

  Map<String, dynamic> toJson() => {
        'startingHours': startingHours,
        'startingMinutes': startingMinutes,
        'endingHours': endingHours,
        'endingMinutes': endingMinutes,
      };
}

File: packages/motion_sleep/ios/Classes/Entities/SleepTime.swift
//
//  SleepTime.swift
//  motion_sleep
//
//  Created by Michael Jajou on 2/23/23.
//

import Foundation

struct SleepTime {
    var startingHours: Int
    var startingMinutes: Int
    var endingHours: Int
    var endingMinutes: Int
    
    static func fromDictionary(_ data: NSDictionary) -> SleepTime {
        return SleepTime(
            startingHours: data["startingHours"] as! Int,
            startingMinutes: data["startingHours"] as! Int,
            endingHours: data["startingHours"] as! Int,
            endingMinutes: data["startingHours"] as! Int
        )
    }
}
